src/view/hooks/useUploadImage.ts

```diff
@@ -0,0 +1,201 @@
+                    throw new Error("downloadUrl not found");
+                }
+
+                return {
+                    userId,
+                    placeId,
+                    photoUrl: uploadTaskResultOfFile.downloadUrl,
+                    width: fileWithSize.size.width,
+                    height: fileWithSize.size.height,
+                };
+            });
+            dispatch(uploadPlacePhotosInPlan({ planId, photos: photos }));
+
+            toast({
+                title: "画像のアップロードが完了しました",
+                status: "success",
+                duration: 3000,
+                isClosable: true,
+            });
+            setUploadRequestStatus(UploadRequestStatus.FULFILLED);
+        } catch (error) {
+            toast({
+                title: "画像のアップロードに失敗しました",
+                description: "もう一度試してみてください。",
+                status: "error",
+                duration: 5000,
+                isClosable: true,
+            });
+            setUploadRequestStatus(UploadRequestStatus.REJECTED);
+        }
+    };
+
+    const setupUploadTaskListener = (
+        taskIndex: number,
+        uploadTask: UploadTask
+    ): Promise<{ taskIndex?: number; downloadUrl: string }> => {
+        return new Promise((resolve, reject) => {
+            uploadTask.on(
+                "state_changed",
+                () => {
+                    // アップロードの進捗が変化した場合の処理
+                },
+                (error) => {
+                    reject(error);
+                },
+                async () => {
+                    try {
+                        const downloadUrl = await getDownloadURL(
+                            uploadTask.snapshot.ref
+                        );
+                        setUploadRequestStatus(UploadRequestStatus.FULFILLED);
+                        resolve({ taskIndex, downloadUrl });
+                    } catch (error) {
+                        reject(error);
+                    }
+                }
+            );
+        });
+    };
+
+    return {
+        localFiles,
+        localImageURLs,
+        uploadedImageURLs,
+        isUploading: uploadRequestStatus === UploadRequestStatus.PENDING,
+        isUploadConfirmationDialogVisible: localFiles.length > 0,
+        handleFileChange,
+        handleUpload,
+        dialogVisible,
+        setDialogVisible,
+    };
+};
+
+// 画像の高さと横幅を取得する
+function fetchImageSizeFromFile(
+    file: File
+): Promise<{ width: number; height: number }> {
+    return new Promise((resolve, reject) => {
+        const image = new Image();
+
+        image.onload = () => {
+            const size = {
+                width: image.naturalWidth,
+                height: image.naturalHeight,
+            };
+            URL.revokeObjectURL(image.src);
+            resolve(size);
+        };
+
+        image.onerror = (error) => {
+            reject(error);
+        };
+
+        image.src = URL.createObjectURL(file);
+    });
+}
+
+export default useUploadImage;
```

Comment:
```suggestion
export const useUploadImage = 
```
とするとこの記述がいらなくなりますよー
（importを修正しないといけないかもだけど）